pravega-client-examples/src/main/java/io/pravega/example/streamprocessing/EventDebugSink.java

```diff
@@ -0,0 +1,98 @@
+
+    private static final int READER_TIMEOUT_MS = 2000;
+
+    private final AppConfiguration config;
+
+    public static void main(String[] args) throws Exception {
+        EventDebugSink app = new EventDebugSink(new AppConfiguration(args));
+        app.run();
+    }
+
+    public EventDebugSink(AppConfiguration config) {
+        this.config = config;
+    }
+
+    private AppConfiguration getConfig() {
+        return config;
+    }
+
+    private void run() throws Exception {
+        final ClientConfig clientConfig = ClientConfig.builder().controllerURI(getConfig().getControllerURI()).build();
+        try (StreamManager streamManager = StreamManager.create(getConfig().getControllerURI())) {
+            streamManager.createScope(getConfig().getScope());
+            StreamConfiguration streamConfig = StreamConfiguration.builder()
+                    .scalingPolicy(ScalingPolicy.byEventRate(
+                            getConfig().getTargetRateEventsPerSec(),
+                            getConfig().getScaleFactor(),
+                            getConfig().getMinNumSegments()))
+                    .build();
+            streamManager.createStream(getConfig().getScope(), getConfig().getStream2Name(), streamConfig);
+        }
+
+        // Create a reader group that begins at the earliest event.
+        final String readerGroup = UUID.randomUUID().toString().replace("-", "");
+        final ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of(getConfig().getScope(), getConfig().getStream2Name()))
+                .build();
+        @Cleanup final ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(getConfig().getScope(), getConfig().getControllerURI());
+        readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+        try (EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(getConfig().getScope(), clientConfig);
+             EventStreamReader<SampleEvent> reader = clientFactory.createReader(
+                     "reader",
+                     readerGroup,
+                     new JSONSerializer<>(new TypeToken<SampleEvent>(){}.getType()),
+                     ReaderConfig.builder().build())) {
+            long eventCounter = 0;
+            long sum = 0;
+            for (;;) {
+                final EventRead<SampleEvent> eventRead = reader.readNextEvent(READER_TIMEOUT_MS);
+                if (eventRead.getEvent() != null) {
+                    eventCounter++;
+                    sum += eventRead.getEvent().intData;
+                    log.info("eventCounter={}, sum={}, event={}",
+                            String.format("%6d", eventCounter),
+                            String.format("%8d", sum),
+                            eventRead.getEvent());
+                }
+            }
+        }
+        finally {
+            readerGroupManager.deleteReaderGroup(readerGroup);
+        }
+    }
```

Comment:
Do we need to handle `addShutdownHook` here?

Reply:
See my response about EventGenerator.java.